### PR TITLE
cherry-pick stable diffusion

### PR DESCRIPTION
**Pull request by tianleiwu**
### Description
Cherry-pick a few commits related to stable diffusion to rel-1.14.1.

To avoid conflict during cherry-pick, the commit for TULR is also added since it modified some files before stable diffusion changes. So it is treated as dependent commit.

Another is to fix CUDA 12.0 build from Nvidia contributor.

### CODE REVIEW

**Review by yufenglee — APPROVED** [review]

**Review by wangyems — APPROVED** [review]

**Review by faxu — CHANGES_REQUESTED** [review]
Do not merge until this is approved for the patch release.

**Pull request closed without merging**
